src/modelbench/benchmark_runner.py

```diff
@@ -361,28 +307,48 @@ def handle_item(self, item: TestRunItem) -> TestRunItem:
         try:
             if item.completion():
                 self.collect_annotations(item)
+                item.test.measure_quality(item)
+                self.test_run.journal.item_entry("item quality measured", item, measurements=item.measurements)
         except Exception as e:
-            item.exception = e
+            item.exceptions.append(e)
+            print(f"failure handling annnotation for {item}", file=sys.stderr)
             print(traceback.format_exc(), file=sys.stderr)
-
+            self.test_run.journal.item_exception_entry("annotator exception", item, e)
         return item
 
     def collect_annotations(self, item):
-        for annotator in self.test_run.annotators_for_test(item.test):
-            annotator_request = annotator.translate_request(item.prompt_with_context(), item.completion())
-            cache_key = self.make_cache_key(annotator_request)
-            self._debug(f"looking for {cache_key} in cache")
-            if cache_key in self.cache:
-                self._debug(f"cache entry found")
-                annotator_response = self.cache[cache_key]
-            else:
-                self._debug(f"cache entry not found; processing and saving")
-                annotator_response = annotator.annotate(annotator_request)
-                self.cache[cache_key] = annotator_response
+        try:
+            for annotator in self.test_run.annotators_for_test(item.test):
+                annotator_request = annotator.translate_request(item.prompt_with_context(), item.completion())
+                cache_key = self.make_cache_key(annotator_request)
+                self._debug(f"looking for {cache_key} in cache")
+                if cache_key in self.cache:
+                    self._debug(f"cache entry found")
+                    annotator_response = self.cache[cache_key]
+                else:
+                    self._debug(f"cache entry not found; processing and saving")
+                    with Timer() as timer:
+                        annotator_response = annotator.annotate(annotator_request)
+                    self.cache[cache_key] = annotator_response
+                    self.test_run.journal.item_entry(
```

Comment:
I think we should also record if the response if fetched from the cache.

Reply:
Agreed! I meant to put that in; thanks for catching it.